File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  attr_reader :sum_of_squares
  attr_reader :square_of_sum
  attr_reader :difference_of_squares
  def initialize(num)
    @sum_of_squares = 0
    sum = 0
    for i in 1..num
      @sum_of_squares += i**2
      sum += i
    end
    @square_of_sum = sum ** 2
    @difference_of_squares = @square_of_sum - @sum_of_squares
  end
end

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number){
    unsigned int sum = 0;
    for (unsigned int i = 1; i <= number; i++)
    {
        sum += i*i;
    }
    return sum;
}

unsigned int square_of_sum(unsigned int number){
    return number*number*(number + 1)*(number + 1)/4;
}

unsigned int difference_of_squares(unsigned int number){
    unsigned int sum = 0;
    for (unsigned int i = 1; i < number + 1; i++)
    {
        for (unsigned int j = 1; j < i + 1; j++)
        {
            if (i != j) sum += 2*i*j;
        }
    }
    return sum;
}